alma/controllers/admin/AdminAlmaRefunds.php

```diff
@@ -109,25 +82,20 @@ public function ajaxProcessRefund()
             $this->ajaxFail(
                 $this->module->l('There was an error while processing the refund', 'AdminAlmaRefunds')
             );
-        } else {
-            $fees = $refundResult->customer_fee;
-            $totalOrder = $refundResult->purchase_amount + $fees;
-            $totalOrderAmount = almaFormatPrice($totalOrder, (int) $order->id_currency);
-            foreach ($refundResult->refunds as $refund) {
-                $totalRefund += $refund->amount;
-            }
-            $totalRefundAmount = almaFormatPrice($totalRefund, (int) $order->id_currency);
-            $percentRefund = (100 / $totalOrder) * $totalRefund;
         }
+        $totalOrder = $refundResult->purchase_amount;
+        $totalOrderAmount = almaFormatPrice($totalOrder, (int) $order->id_currency);
+        foreach ($refundResult->refunds as $refund) {
+            $totalRefund += $refund->amount;
+        }
+        if ($totalRefund > $totalOrder) {
+            $totalRefund = $totalOrder;
+        }
+        $totalRefundAmount = almaFormatPrice($totalRefund, (int) $order->id_currency);
+        $percentRefund = (100 / $totalOrder) * $totalRefund;
```

Comment:
can you add methods for each required variable:

* `buildTotalRefund()`
* `buildTotalOrderAmount()`
* `buildTotalRefundAmount()`
* `buildPercentRefund()`

What is the difference between `$totalRefund` and `$totalRefundAmount`,   (the same for `$totalOrder` and `$totalOrderAmount`) ... is there a better name to give to these variables / methods ?

Reply:
@syjust-alma `$totalRefund` is the number (2000) and `$totalRefundAmount` is the string (20,00 €) same for `$totalOrder`.